providers/packetframe/packetframeProvider.go

```diff
@@ -0,0 +1,245 @@
+package packetframe
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/StackExchange/dnscontrol/v3/models"
+	"github.com/StackExchange/dnscontrol/v3/pkg/diff"
+	"github.com/StackExchange/dnscontrol/v3/providers"
+)
+
+// packetframeProvider is the handle for this provider.
+type packetframeProvider struct {
+	client      *http.Client
+	baseURL     *url.URL
+	token       string
+	domainIndex map[string]zone
+}
+
+var defaultNameServerNames = []string{
+	"ns1v4.packetframe.com",
+	"ns2v4.packetframe.com",
+}
+
+// NewPacketframe creates the provider.
+func NewPacketframe(m map[string]string, metadata json.RawMessage) (providers.DNSServiceProvider, error) {
```

Comment:
`NewPacketframe` does not need to be exported and can be renamed to `newPacketframe`